File: Easy/2169_Count_Operations_to_Obtain_Zero/Solution.java


public class Solution {
    public int countOperations(int num1, int num2) {
        int stap = 0;

        while (num1 != 0 && num2 != 0) {
            if (num1 >= num2) {
                num1 -= num2;
            } else {
                num2 -= num1;
            }

            stap++;
        }

        return stap;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        System.out.println(solution.countOperations(2, 3));

        // test cases 2
        System.out.println(solution.countOperations(10, 10));
        
    }
}

File: Easy/2169_Count_Operations_to_Obtain_Zero/Solution.py
from typing import List, Optional


class Solution:
    def countOperations(self, num1: int, num2: int) -> int:
        count = 0

        while num1 > 0 and num2 > 0:
            if num1 >= num2:
                num1 -= num2
            else:
                num2 -= num1
            count += 1
        
        return count
        


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    print(solution.countOperations(2, 3))

    # test cases 2
    print(solution.countOperations(10, 10))
    